src/test/regress/expected/multi_mx_master_update_node.out

```diff
@@ -0,0 +1,373 @@
+-- Test creation of mx tables and metadata syncing
+SELECT nextval('pg_catalog.pg_dist_placement_placementid_seq') AS last_placement_id
+\gset
+SELECT nextval('pg_catalog.pg_dist_groupid_seq') AS last_group_id \gset
+SELECT nextval('pg_catalog.pg_dist_node_nodeid_seq') AS last_node_id \gset
+SELECT nextval('pg_catalog.pg_dist_colocationid_seq') AS last_colocation_id \gset
+SELECT nextval('pg_catalog.pg_dist_shardid_seq') AS last_shard_id \gset
+SET citus.replication_model TO streaming;
+SET citus.shard_count TO 8;
+SET citus.shard_replication_factor TO 1;
+-- set sync intervals to less than 5s so wait_until_metadata_sync never times out
+ALTER SYSTEM SET citus.metadata_sync_interval TO 4000;
+ALTER SYSTEM SET citus.metadata_sync_retry_interval TO 1000;
+SELECT pg_reload_conf();
+ pg_reload_conf 
+----------------
+ t
+(1 row)
+
+CREATE FUNCTION wait_until_metadata_sync(timeout INTEGER DEFAULT 5000)
+    RETURNS void
+    LANGUAGE C STRICT
+    AS 'citus';
+-- Verifies pg_dist_node and pg_dist_palcement in the given worker matches the ones in coordinator
+CREATE FUNCTION verify_metadata(hostname TEXT, port INTEGER, master_port INTEGER DEFAULT 57636)
+    RETURNS BOOLEAN
+    LANGUAGE sql
+    AS $$
+WITH dist_node_summary AS (
+    SELECT 'SELECT jsonb_agg(ROW(nodeid, groupid, nodename, nodeport, isactive) ORDER BY nodeid) FROM  pg_dist_node' as query
+), dist_node_check AS (
+    SELECT count(distinct result) = 1 AS matches
+    FROM dist_node_summary CROSS JOIN LATERAL
+        master_run_on_worker(ARRAY[hostname, 'localhost'], ARRAY[port, master_port],
+                            ARRAY[dist_node_summary.query, dist_node_summary.query],
+                            false)
+), dist_placement_summary AS (
+    SELECT 'SELECT jsonb_agg(pg_dist_placement ORDER BY shardid) FROM pg_dist_placement)' AS query
+), dist_placement_check AS (
+    SELECT count(distinct result) = 1 AS matches
+    FROM dist_placement_summary CROSS JOIN LATERAL
+        master_run_on_worker(ARRAY[hostname, 'localhost'], ARRAY[port, master_port],
+                            ARRAY[dist_placement_summary.query, dist_placement_summary.query],
+                            false)
+)
+SELECT dist_node_check.matches AND dist_placement_check.matches
+FROM dist_node_check CROSS JOIN dist_placement_check
+$$;
+-- Simulates a readonly node by setting default_transaction_read_only. 
+CREATE FUNCTION mark_node_readonly(hostname TEXT, port INTEGER, isreadonly BOOLEAN)
+    RETURNS TEXT
+    LANGUAGE sql
+    AS $$
+    SELECT master_run_on_worker(ARRAY[hostname], ARRAY[port],
+           ARRAY['ALTER SYSTEM SET default_transaction_read_only TO ' || isreadonly::TEXT], false);
+    SELECT result FROM
+        master_run_on_worker(ARRAY[hostname], ARRAY[port],
+                             ARRAY['SELECT pg_reload_conf()'], false);
+$$;
+-- add a node to the cluster
+SELECT master_add_node('localhost', :worker_1_port) As nodeid_1 \gset
+SELECT * FROM pg_dist_node;
```

Comment:
This will probably break when we add more columns.

Reply:
done

Reply:
Also updated multi_read_from_secondaries to not fail by changes to pg_dist_node. I think it should be good to see the whole pg_dist_node entries in multi_cluster_management though.